lattice_bscan_spi.py

```diff
@@ -149,6 +149,7 @@ def elaborate(self, platform):
         #   JTAG adapter: sample TDO
         with m.FSM() as fsm:
             with m.State("IDLE"):
+                m.d.sync += self.cs_n.o.eq(1)
```

Comment:
Since the Output Enable for the tri-stated CS_N pin is already set to 1 whenever the JTAG chain 1 has been selected ([lines 97](https://github.com/quartiq/bscan_spi_bitstreams/blob/fe3a91a48fd207da2dd681dfc6332727bea54eb6/lattice_bscan_spi.py#L97) and [130](https://github.com/quartiq/bscan_spi_bitstreams/blob/fe3a91a48fd207da2dd681dfc6332727bea54eb6/lattice_bscan_spi.py#L130)), and the Output value of this pin is automatically assigned to 0 whenever the SPI transfer is ongoing ([line 167](https://github.com/quartiq/bscan_spi_bitstreams/blob/fe3a91a48fd207da2dd681dfc6332727bea54eb6/lattice_bscan_spi.py#L167)), I don't think these changes to the CS_N pin (this line and your line 163) are necessary.